asus: check for hid report and fix chakram x

kyokenn opened the pull request:
Fixes https://github.com/libratbag/libratbag/issues/1413

Every ASUS mouse have multiple interfaces for mouse, keyboard, configuration and other stuff. There was some error messages because libratbag tried to initialize on the wrong interfaces of the mouse. It tries every interface and fails on the wrong one with error message. This PR fixes it and I have tested this part.

staticssleever668 commented on src/driver-asus.c:
```diff
@@ -443,6 +443,11 @@ asus_driver_probe(struct ratbag_device *device)
+	/* HID device descriptor must have input and output */
+	if (!ratbag_hidraw_has_report(device, HID_INPUT_REPORT) ||
+	    !ratbag_hidraw_has_report(device, HID_OUTPUT_REPORT))
+		return -1;
```
Could you change this to `-ENODEV`?
The `probe` function is supposed to only return POSIX error codes on error, moreover, if it's something other than ENODEV it's going to print it to stderr.

staticssleever668 reviewed (changes requested):
One little change and it's good (you could also split this into two commits but whatever). Thank you so much for your work on the asus driver! :)

kyokenn commented:
"ratbag_hidraw_has_report" doesn't work for me anymore for some reason. Maybe I have updated some packages. So I replaced it with profile request, it should work even when the mouse is sleeping or turned off.

staticssleever668 commented:
> "ratbag_hidraw_has_report" doesn't work for me anymore for some reason.
Well, if you find a way to use it instead, feel free to open another PR to avoid duplicate work.
